Sources/BraintreePayPal/BTPayPalClient.swift

```diff
@@ -389,7 +389,7 @@ import BraintreeDataCollector
             return
         }
 
-        application.open(redirectURL, options: [:]) { success in
+        application.openURL(redirectURL, options: [:]) { success in
```

Comment:
The method name was updated since the method in the protocol `URLOpener` no longer exists on `UIApplication` in Xcode 16 beta 3. This allows us to under the hood call the correct `open` method on `UIApplication` in both lower/CI version of Xcode as well as the updates in Xcode 16 beta 3+.

Reply:
Ah, so it looks like in line UIApplication+URLOpener line 41, we're always calling into the older method `UIApplication.shared.open()`. 

Should we instead use the newer one, `UIApplication.shared.openURL()`, if it's available? I am just downloading the new beta now, but is there a difference b/w the 2?

Reply:
So the issue is that the method signature changed _only_ in Xcode 16 beta 3 ([documented by Apple here](https://developer.apple.com/documentation/uikit/uiapplication/1648685-open?changes=latest_minor) in the `API changes`). This new signature does not exist in Xcode 16 beta 2 and below or Xcode 15. 

How we have the protocol `URLOpener` we needed no implementation for `canOpenURL` or `open` in any version of Xcode below 16 beta 3 because our signatures matched the signature for these methods in `UIApplication` so no implementation was needed in our extension of this class that conforms to our `URLOpener` protocol.

Now because the signature changed we need to handle both the old and new signatures. This PR updates to now just create a new protocol method in `URLOpener` that under the hood calls the correct `open` method for older and new version of the signature but we cannot use the new signature in older version or the old signature in newer versions.

Reply:
Ah I see! Thanks for clarifying. Feels unusual for Apple to change a method without deprecating it first

Reply:
I guess once Xcode 16 is the minimum we support, we can remove this implementation in the UIApplication extension entirely. We can go back to the way it was, using the new method name. Should we leave a note for that?

Reply:
No worries! It's a weird one for sure, was a bit odd to find they made that change randomly in a beta without much more indication. Agree we can move back to the how it was implemented previously. I can add a `TODO` for this.